pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -1671,24 +1684,29 @@ func updateRootSnapshots(
 	}
 
 	// Refresh the root snapshots from the VM mo
-	var newRootSnapshots []common.LocalObjectRef //nolint:prealloc
+	var newRootSnapshots []vmopv1.VirtualMachineSnapshotReference //nolint:prealloc
 	for _, rootSnapshot := range vmCtx.MoVM.Snapshot.RootSnapshotList {
 		rootSnapshotCR, err := getSnapshotCR(vmCtx, k8sClient, rootSnapshot.Name)
 		if err != nil {
 			return err
 		}
 
 		if rootSnapshotCR == nil {
-			// TODO (Lubron): This is for external snapshots.
-			vmCtx.Logger.V(4).Info("Snapshot custom resource not found, skipping",
+			vmCtx.Logger.V(4).Info("Snapshot custom resource not found, adding an Unmnaged reference",
 				"snapshotName", rootSnapshot.Name)
+			newRootSnapshots = append(newRootSnapshots, vmopv1.VirtualMachineSnapshotReference{
```

Comment:
Care there be multiple of these? Is there something later set to help identify which is which?

Reply:
Yes, it's possible that multiple of these are there. There isn't an info to identify them.

We discussed about having the snap moID as a field in `VirtualMachineSnapshotReference` when the snapshot is not managed, but we didn't find an utility of having it at the moment.

When someone has snapshots in VC which are not managed by VMOP, we don't support operations like reverting to those snapshots.